### PR TITLE
Apm 333242 app engine not created when running install

### PR DESCRIPTION
**Pull request by karolkieglerski**

### CODE REVIEW

**Review by mswiatkowska — APPROVED** [review]

File: scripts/setup.sh [hunk 1]
```diff
@@ -222,9 +206,55 @@ gcloud config set project $GCP_PROJECT
...
+      exit
+      ;;
+    *) echo "- please answer y or n" ;;
+    esac
+  done
+elif [[ "$SERVING_APP_ENGINE" != "SERVING"  ]]; then
+  echo
+  echo -e "\e[91mERROR: \e[37mTo continue deployment your GCP project must contain an App Engine app - it's required for Cloud Scheduler"
+  echo
+  echo 'Enable App Engine on: https://console.cloud.google.com/appengine/settings?_ga=2.67541841.1390822090.1635945824-1034520987.1633950930'
```
Comment (sq2gxo) [inline scripts/setup.sh:255]: should this link be shorter?
https://console.cloud.google.com/appengine/settings

**Review by mswiatkowska — DISMISSED** [review]

File: scripts/setup.sh [hunk 2]
```diff
@@ -14,66 +14,61 @@
...
-if ! command -v gcloud &> /dev/null
-then
-
-    echo -e "\e[91mERROR: \e[37mGoogle Cloud CLI is required to install Dynatrace function. Go to following link in your browser and download latest version of Cloud SDK:"
-    echo -e
-    echo -e "https://cloud.google.com/sdk/docs#install_the_latest_cloud_tools_version_cloudsdk_current_version"
-    echo -e
-    echo
-    exit
+  echo -e "\e[91mERROR: \e[37mGoogle Cloud CLI is required to install Dynatrace function. Go to following link in your browser and download latest version of Cloud SDK:"
```
Comment (Dawid-Kaszubski) [inline scripts/setup.sh:60]: Minor: It might be printed with `err` function

File: scripts/setup.sh [hunk 3]
```diff
@@ -14,66 +14,61 @@
...
+  exit
 fi
 
-if ! command -v unzip &> /dev/null
-then
-    echo -e "\e[91mERROR: \e[37munzip is required to install Dynatrace function"
-    echo
-    exit
+if ! command -v unzip &>/dev/null; then
+  echo -e "\e[91mERROR: \e[37munzip is required to install Dynatrace function"
```
Comment (Dawid-Kaszubski) [inline scripts/setup.sh:69]: Minor: It might be printed with err function

File: tests/e2e/metrics/test_metrics.py [hunk 4]
```diff
@@ -43,4 +43,5 @@ def test_metrics_on_dynatrace():
     response_json = response.json()
     assert 'totalCount' in response_json
     assert response_json['totalCount'] == 1
+    print(response_json)
```
Comment (Dawid-Kaszubski) [inline tests/e2e/metrics/test_metrics.py:46]: Should it be in this PR? 😄

File: scripts/setup.sh (hunk 2 above)
Comment (karolkieglerski) [reply scripts/setup.sh:60]: @Dawid-Kaszubski on master we don't have `err` function

File: scripts/setup.sh (hunk 3 above)
Comment (karolkieglerski) [reply scripts/setup.sh:69]: the same

File: tests/e2e/metrics/test_metrics.py (hunk 4 above)
Comment (karolkieglerski) [reply tests/e2e/metrics/test_metrics.py:46]: yes

File: scripts/setup.sh (hunk 2 above)
Comment (Dawid-Kaszubski) [reply scripts/setup.sh:60]: Right! We use it a couple of lines above, so I've assumed that is added.

File: scripts/setup.sh [hunk 5]
```diff
@@ -14,66 +14,61 @@
...
-if ! command -v yq &> /dev/null
-then
+if ! command -v yq &> /dev/null; then
 
-    echo -e "\e[91mERROR: \e[37m yq and jq is required to install Dynatrace function. Please refer to following links for installation instructions:"
-    echo -e
-    echo -e "YQ: https://github.com/mikefarah/yq"
-    if ! command -v jq &> /dev/null
-    then
+    err 'yq and jq is required to install Dynatrace function. Please refer to following links for installation instructions:
```
Comment (Dawid-Kaszubski) [inline scripts/setup.sh:32]: Please change to `echo`

File: scripts/setup.sh [hunk 6]
```diff
@@ -14,66 +14,61 @@
...
+    Example command to install yq:
+    sudo wget https://github.com/mikefarah/yq/releases/download/v4.9.8/yq_linux_amd64 -O /usr/bin/yq && sudo chmod +x /usr/bin/yq'
+    if ! command -v jq &> /dev/null; then
         echo -e "JQ: https://stedolan.github.io/jq/download/"
     fi
-    echo -e
-    echo -e "You may also try installing YQ with PIP: pip install yq"
-    echo -e ""
-    echo
+    err 'You may also try installing YQ with PIP: pip install yq'
```
Comment (Dawid-Kaszubski) [inline scripts/setup.sh:39]: Same, please change to `echo`

**Review by Dawid-Kaszubski — APPROVED** [review]

File: tests/e2e/metrics/test_metrics.py (hunk 4 above)
Comment (karolkieglerski) [reply tests/e2e/metrics/test_metrics.py:46]: this will show full response when test will fail

**Review by Dawid-Kaszubski — APPROVED** [review]

**Review by dtwwolinski — DISMISSED** [review]

**Review by mswiatkowska — APPROVED** [review]

**Review by sq2gxo — APPROVED** [review]

**Pull request merged**